config/application.rb

```diff
@@ -194,6 +194,12 @@ module ::Reporting; end
     config.i18n.available_locales = OpenFoodNetwork::I18nConfig.available_locales
     I18n.locale = config.i18n.locale = config.i18n.default_locale
 
+    # Calculate digests for locale files so we can know when they change
+    config.i18n.available_locales.each do |locale|
+      (config.x.i18n_digests ||= {})[locale.to_sym] =
+        Digest::MD5.hexdigest(File.read(Rails.root.join("config/locales/#{locale}.yml")))
```

Comment:
This would be worth expanding into a few extra lines, to help mere mortals such as myself to understand it. 

In what cases is `config.x.i18n_digests` already present, and why do we need to overwrite it?

Reply:
I've updated this after Maikel's comment, removing the `||=` part.

`config.x.i18n_digests` is essentially an empty hash when the app initializes. For each enabled locale (set in `I18n.available_locales`) we're adding a key to that hash, and the value of that key will be an MD5 digest of the content of the relevant locale file.

So if `I18n.available_locales` is `[:en, :en_GB, :fr]`, we'll be setting `config.x.i18n_digests` to something like this when the app initializes:
```ruby
{
  en: "a9e8d7ea9d87e", # MD5 of the contents of config/locales/en.yml
  en_GB: "ef08c79d87d9c7", # MD5 of the contents of config/locales/en_GB.yml
  fr: "98df7df80c098d" # MD5 of the contents of config/locales/fr.yml
}
```

Next time the app initializes, if the contents of `config/locales/en_GB.yml` are different, it's key will be different so any cache keys derived from that value will automatically expire.